src/libraries/System.Text.Json/System.Text.Json.SourceGeneration.UnitTests/JsonSourceGeneratorTests.cs

```diff
@@ -85,66 +88,57 @@ public class MyType {
 
                     public void MyMethod() { }
                     public void MySecondMethod() { }
+                    public void UsePrivates()
+                    {
+                        PrivateDouble = 0;
+                        PrivateChar = ' ';
+                        double x = PrivateDouble;
+                        string s = PrivateChar.ToString();
+                    }
                 }
 
                 [JsonSerializable(typeof(JsonConverterAttribute))]
                 public class NotMyType { }
-
               }";
 
             Compilation compilation = CompilationHelper.CreateCompilation(source);
 
             JsonSerializerSourceGenerator generator = new JsonSerializerSourceGenerator();
 
-            Compilation outCompilation = CompilationHelper.RunGenerators(compilation, out var generatorDiags, generator);
+            Compilation newCompilation = CompilationHelper.RunGenerators(compilation, out ImmutableArray<Diagnostic> generatorDiags, generator);
+
+            // Make sure compilation was successful.
+            Assert.Empty(generatorDiags);
+            Assert.Empty(newCompilation.GetDiagnostics());
```

Comment:
Should we have similar debug asserts for the equivalent areas in the actual generator? i.e. expressing in the code that we are assuming that the compilation of the input context was successful.

Reply:
No, in the actual generator compilation will fail and it will tell the user. The user is also notified that source generated content is within their compilation. Don't think there's an explicit need to do it through a debug assert.